Sources/Defaults/Defaults+iCloud.swift

```diff
@@ -0,0 +1,571 @@
+			valueDescription = " with value \(value ?? "nil") "
+		case .completed:
+			status = "Complete synchronization"
+		}
+
+		let message = "\(status) key '\(key.name)'\(valueDescription)\(destination)"
+		log(message)
+	}
+
+	private static func log(_ message: String) {
+		guard Defaults.iCloud.isDebug else {
+			return
+		}
+
+		if #available(macOS 11, iOS 14, tvOS 14, watchOS 7, *) {
+			logger.debug("[Defaults.iCloud] \(message)")
+		} else {
+			#if canImport(OSLog)
+			os_log(.debug, log: .default, "[Defaults.iCloud] %@", message)
+			#else
+			let dateFormatter = DateFormatter()
+			dateFormatter.dateFormat = "yyyy-MM-dd HH:mm:ss.SSSSSSZZZ"
+			let dateString = dateFormatter.string(from: Date())
+			let processName = ProcessInfo.processInfo.processName
+			let processIdentifier = ProcessInfo.processInfo.processIdentifier
+			var threadID: UInt64 = 0
+			pthread_threadid_np(nil, &threadID)
+			print("\(dateString) \(processName)[\(processIdentifier):\(threadID)] [Defaults.iCloud] \(message)")
+			#endif
+		}
+	}
+}
+
+extension Defaults {
+	/**
+	Represent different data sources available for synchronization.
+	*/
+	public enum DataSource {
+		/**
+		Using `key.suite` as data source.
+		*/
+		case local
+
+		/**
+		Using `NSUbiquitousKeyValueStore` as data source.
+		*/
+		case remote
+	}
+
+	/**
+	Synchronize values with different devices over iCloud.
+
+	There are four ways to initiate synchronization, each of which will create a task in `backgroundQueue`:
+
+	1. Using ``iCloud/add(_:)-5gffb``
+	2. Utilizing ``iCloud/syncWithoutWaiting(_:source:)-9cpju``
+	3. Observing UserDefaults for added ``Defaults/Defaults/Key`` using Key-Value Observation (KVO)
+	4. Monitoring `NSUbiquitousKeyValueStore.didChangeExternallyNotification` for added ``Defaults/Defaults/Key``.
+
+	> Tip: After initializing the task, we can call ``iCloud/sync()`` to ensure that all tasks in the backgroundQueue are completed.
+
+	```swift
+	import Defaults
+
+	extension Defaults.Keys {
+		static let isUnicornMode = Key<Bool>("isUnicornMode", default: true, iCloud: true)
+	}
+
+	Task {
+		let quality = Defaults.Key<Int>("quality", default: 0)
+		Defaults.iCloud.add(quality)
+		await Defaults.iCloud.sync() // Using sync to make sure all synchronization tasks are done.
+		// Both `isUnicornMode` and `quality` are synced.
+	}
+	```
+	*/
+	public enum iCloud {
+		/**
+		The singleton for Defaults's iCloudSynchronizer.
+		*/
+		static var synchronizer = iCloudSynchronizer(remoteStorage: NSUbiquitousKeyValueStore.default)
+
+		/**
+		Lists the synced keys.
+		*/
+		public static var keys: Set<Defaults.Keys> { synchronizer.keys }
+
+		/**
+		Enable this if you want to call ```` when a value is changed.
+		*/
+		public static var syncOnChange = false
+
+		/**
+		Enable this if you want to debug the syncing status of keys.
+
+		- Note: The log information will include details such as the key being synced, its corresponding value, and the status of the synchronization.
```

Comment:
I meant that it's not clear that `isDebug` causes log info to be printed to the console. That what should be clarified.